### PR TITLE
Convert docs to jupyter-book.

### PR DESCRIPTION
**Pull request by grahamgower**
I probably missed something, but this should take care of most things.

### CODE REVIEW

File: demes/demes.py [hunk 1]
```diff
@@ -1877,7 +1870,7 @@ def discrete_demographic_events(self) -> Dict[str, List[Any]]:
             )
         return demo_events
 
-    def in_generations(self):
+    def in_generations(self) -> "Graph":
```
Comment (jeromekelleher) [inline demes/demes.py:1873]: Maybe use the ``from __future__`` thing so you can write it without the quotes? Or is that only available in 3.7+?

File: docs/_config.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    - sphinx.ext.intersphinx
+    - sphinx_issues
+
+    config:
+      issues_github_path: popsim-consortium/demes-python
+      todo_include_todos: true
+      intersphinx_mapping:
+        python: ["https://docs.python.org/3/", null]
+        spec: ["https://popsim-consortium.github.io/demes-spec-docs/main/", null]
+        stdpopsim: ["https://stdpopsim.readthedocs.io/en/stable", null]
```
Comment (jeromekelleher) [inline docs/_config.yml:41]: Do we need stdpopsim and msprime intersphinx mappings? We shouldn't really be depending on their APIs, right?

File: docs/changelog.md [hunk 3]
```diff
@@ -1,5 +1,5 @@
-=========
-Changelog
-=========
+# Changelog
 
+```{eval-rst}
 .. include:: ../CHANGELOG.rst
```
Comment (jeromekelleher) [inline docs/changelog.md:4]: Might be as easy update the CHANGELOG to markdown now. It'll make it easier to work with in GitHub.

What I did for msprime is put in a soft link to the CHANGELOG.md file in the project root here. Works fine.

**Review by jeromekelleher — APPROVED** [review]
Fantastic! Minor comments.

File: demes/demes.py (hunk 1 above)
Comment (grahamgower) [reply demes/demes.py:1873]: Unfortunately that was introduced in 3.7. BTW, that future becomes the default behaviour in 3.10.

File: docs/_config.yml (hunk 2 above)
Comment (grahamgower) [reply docs/_config.yml:41]: Just for the `demes.convert` functions. How about I just remove that docs page now though?

Comment (jeromekelleher) [reply docs/_config.yml:41]: Yes, I think so. There's no point in documenting something we expect to remove immediately and don't want people to use.

**Pull request merged**